expression/builtin_time.go

```diff
@@ -638,15 +638,16 @@ func builtinDateArith(args []types.Datum, ctx context.Context) (d types.Datum, e
 	// Op is used for distinguishing date_add and date_sub.
 	// args[0] -> Op
```

Comment:
If you like, you can remove the args[0], and use a builtin function factory instead. Like what the `compareFuncFactory` do, then we will not rely on any ugly things any more :)